models/DataObject/ClassDefinition/Data.php

```diff
@@ -664,13 +678,22 @@ protected function getPreGetValueHookCode(string $key): string
     public function getGetterCode($class)
     {
         $key = $this->getName();
+
+
+        if ($class->getUseTypeHints() && $this->getReturnType()) {
+            $typehint = ': ' . $this->getReturnType();
```

Comment:
This will intruduce a crash:y scenario on Table data types, which have declared an `array` docblock return type, but in reality will return either a string (i.e. no data) or an array. Either we make a BC break on the Table datatype by making it only return an array, or we can change the return type to `array|string` and validate here if the datatype has a pipe char and not append it as a typehint. See #6532, thoughts?

Reply:
this is only the default return type. it is up to the datatype to generate a better/different return type hint.

Reply:
But as long as are we're not on PHP 8 with union types, it's not possible to add the return type of `array|string` for the mentioned method, so we would have to skip it. Is this supported?

Reply:
@jdreesen Relation types already return pipe-concatenated types, for example: [1](https://github.com/pimcore/pimcore/blob/c485e8d4ea1de0d7bb97fd06331205cce54cede9/models/DataObject/ClassDefinition/Data/ManyToOneRelation.php#L675), [2](https://github.com/pimcore/pimcore/blob/163f95e61ac97b8d5997932827ba93cc85755b0b/models/DataObject/ClassDefinition/Data/ManyToManyObjectRelation.php#L754), [3](https://github.com/pimcore/pimcore/blob/1f9fca52eacf4fcc8b8c7899b65c0b1900db9124/models/DataObject/ClassDefinition/Data/ManyToManyRelation.php#L885). These must also be solved